src/validators/citizenship.js

```diff
@@ -1,7 +1,135 @@
+import _ from 'lodash'
 import NameValidator from './name'
 import LocationValidator from './location'
 import { validGenericTextfield, validDateField } from './helpers'
 
+const validCitizenships = arr => !!arr && arr.length > 0
+
+export const validateAbroadDocumentation = ({ abroadDocumentation, explanation }) => (
+  !!abroadDocumentation
+    && ['FS-240', 'DS-1350', 'FS-545', 'Other'].includes(
+      abroadDocumentation
+    )
+    && (abroadDocumentation !== 'Other'
+      || (abroadDocumentation === 'Other'
+        && validGenericTextfield(explanation)))
+)
+
+export const validateBornOnMilitaryInstallation = ({
+  bornOnMilitaryInstallation,
+  militaryBase,
+}) => !!bornOnMilitaryInstallation
+      && (bornOnMilitaryInstallation === 'No'
+        || (bornOnMilitaryInstallation === 'Yes'
+          && validGenericTextfield(militaryBase)))
+
+export const validateDocumentation = ({
+  abroadDocumentation,
+  explanation,
+  documentNumber,
+  documentIssued,
+  placeIssued,
+  documentName,
+}) => validateAbroadDocumentation({ abroadDocumentation, explanation })
+    && validGenericTextfield(documentNumber)
+    && validDateField(documentIssued)
+    && new LocationValidator(placeIssued).isValid()
+    && new NameValidator(documentName).isValid()
+
+export const validateCertificate = ({
+  certificateNumber,
+  certificateIssued,
+  certificateName,
+}) => validGenericTextfield(certificateNumber)
+    && validDateField(certificateIssued)
+    && new NameValidator(certificateName).isValid()
+
+export const isCertificatePartial = ({ certificateNumber, certificateIssued, certificateName }) => {
+  if (!_.isEmpty(certificateNumber) && !_.isEmpty(certificateNumber.value)) {
+    return !validateCertificate({ certificateNumber, certificateIssued, certificateName })
+  } if (!_.isEmpty(certificateIssued) && (
```

Comment:
This seems like a redundant use of `_.isEmpty` if you have to check each of the object's keys as well